### PR TITLE
Add doxygen/1.8.17

### PR DESCRIPTION
**Pull request by tonka3000**
Specify library name and version:  **doxygen/1.8.17**

This is a port of the bincrafters repo https://github.com/bincrafters/conan-doxygen_installer (stable/1.8.17). I implemented the same tactic as in the nodejs recipe (which is already in the conan-center-index) to download the sources (immutable source).

I've tested it locally on windows 10 x64.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/doxygen/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            tools.check_sha256(filename, sha)
+            if os_name[str(self.settings.os_build)] in url:
+                dest_file = filename
+
+        if not dest_file:
+            raise ConanInvalidConfiguration("could not find source file fo the configuration")
+
+        tools.unzip("doxygen-{}.linux.bin.tar.gz".format(self.version), pattern="*LICENSE")
+
+        if self.settings.os_build == "Macos":
```
Comment (uilianries) [inline recipes/doxygen/all/conanfile.py:63]:
```suggestion
        if self.settings.os == "Macos":
```

File: recipes/doxygen/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 2.8.12)
+cmake_policy(SET CMP0057 NEW) # if IN_LIST
```
Comment (uilianries) [inline recipes/doxygen/all/test_package/CMakeLists.txt:2]: Is it really required? According the documentation, it requires Cmake 3.3

Comment (tonka3000) [reply recipes/doxygen/all/test_package/CMakeLists.txt]: I've changed it to 3.3 and removed the policy

**Comment by uilianries** [comment]
I'm curious why we didn't build from source in Bincrafters. I just tried here and worked pretty well. I didn't check the cmake file, but it required flex and bison.

**Comment by Croydon** [comment]
> I'm curious why we didn't build from source in Bincrafters

I think there are two main reasons:
 
  * This recipe did originate from inexorgame, from a time where there probably were no flex and bison recipes
  * and I'm assuming that this Doxygen build does not have all features, as a complete build requires tons of dependencies which we still don't have today: https://github.com/conan-io/wishlist/issues/115

**Comment by uilianries** [comment]
> and I'm assuming that this Doxygen build does not have all features

True, I check the output and it shows less outputs than for pre-built binaries.

**Pull request merged**